apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/LimitPriceWarning/LimitPriceWarning.tsx

```diff
@@ -0,0 +1,127 @@
+import {useSwap} from '@yoroi/swap'
+import BigNumber from 'bignumber.js'
+import React from 'react'
+import {StyleSheet, Text, View} from 'react-native'
+
+import {Button, Spacer} from '../../../../../../components'
+import {BottomSheetModal} from '../../../../../../components/BottomSheetModal'
+import {useLanguage} from '../../../../../../i18n'
+import {useSelectedWallet} from '../../../../../../SelectedWallet'
+import {useTokenInfo} from '../../../../../../yoroi-wallets/hooks'
+import {useStrings} from '../../../../common/strings'
+
+export interface LimitPriceWarningProps {
+  open: boolean
+  onClose?: () => void
+  onSubmit?: () => void
+}
+
+export const LimitPriceWarning = ({open, onClose, onSubmit}: LimitPriceWarningProps) => {
+  const {createOrder} = useSwap()
+  const {numberLocale} = useLanguage()
+  const strings = useStrings()
+  const limitPrice = BigNumber(createOrder.limitPrice ?? 0).toFormat(numberLocale)
+  const marketPrice = BigNumber(
+    createOrder.selectedPool?.price !== undefined && !Number.isNaN(createOrder.selectedPool.price)
+      ? createOrder.selectedPool.price
+      : 0,
+  ).toFormat(10, numberLocale)
+  const wallet = useSelectedWallet()
+
+  const tokenToSellInfo = useTokenInfo({wallet, tokenId: createOrder.amounts.sell.tokenId})
+  const tokenToSellName = tokenToSellInfo.ticker ?? tokenToSellInfo.name ?? '-'
+  const tokenToBuyInfo = useTokenInfo({wallet, tokenId: createOrder.amounts.buy.tokenId})
+  const tokenToBuyName = tokenToBuyInfo.ticker ?? tokenToBuyInfo.name ?? '-'
+
+  const limitValue = `${limitPrice} ${tokenToSellName}/${tokenToBuyName}`
+  const marketValue = `${marketPrice} ${tokenToSellName}/${tokenToBuyName}`
+
+  return (
+    <BottomSheetModal
```

Comment:
it's product question, but according to cip26, ticker ranges from 2-9 so in the worst case 9/9 (ROBOTokenROBO would be a malformatted token registry entry), the order should be `ticker ?? name ?? assetName` yet it can still become gigantic. Maybe we should limit and add middle ellipses